src/rp2/plugin/report/us/open_positions.py

```diff
@@ -0,0 +1,233 @@
+from rp2.computed_data import ComputedData
+from rp2.in_transaction import InTransaction
+from rp2.logger import create_logger
+from rp2.plugin.report.abstract_ods_generator import AbstractODSGenerator
+from rp2.rp2_decimal import RP2Decimal
+from rp2.rp2_error import RP2TypeError
+
+LOGGER: logging.Logger = create_logger("open_positions")
+
+_TEMPLATE_SHEETS_TO_KEEP: Set[str] = {"__Asset", "__Asset_and_Exchange", "__AssetPrice"}
+_FIAT_EMPTY_BAL_CUTOFF = RP2Decimal("0.01")
+_CRYPTO_EMPTY_BAL_CUTOFF = RP2Decimal(".000001")
+_FIAT_UNIT_WINDOW_2_DECIMAL = RP2Decimal("1")
+_FIAT_UNIT_WINDOW_4_DECIMAL = RP2Decimal("0.02")
+
+
+class Generator(AbstractODSGenerator):
+
+    OUTPUT_FILE: str = "open_positions.ods"
+    HEADER_ROWS = 3
+
+    def generate(
+        self,
+        country: AbstractCountry,
+        accounting_method: str,
+        asset_to_computed_data: Dict[str, ComputedData],
+        output_dir_path: str,
+        output_file_prefix: str,
+        from_date: date,
+        to_date: date,
+    ) -> None:
+
+        # pylint: disable=too-many-branches
+
+        sheet_names: Set[str] = {"Asset", "Asset_and_Exchange", "AssetPrice"}
+        row_indexes: Dict[str, int] = {sheet_name: self.HEADER_ROWS for sheet_name in sheet_names}
+
+        if not isinstance(asset_to_computed_data, Dict):
+            raise RP2TypeError(f"Parameter 'asset_to_computed_data' has non-Dict value {asset_to_computed_data}")
+
+        output_file: Any
+        output_file = self._initialize_output_file(
+            country=country,
+            accounting_method=accounting_method,
+            output_dir_path=output_dir_path,
+            output_file_prefix=output_file_prefix,
+            output_file_name=self.OUTPUT_FILE,
+            template_sheets_to_keep=_TEMPLATE_SHEETS_TO_KEEP,
+            from_date=from_date,
+            to_date=to_date,
+            template_file_prefix="open_positions",
+        )
+
+        asset: str
+        computed_data: ComputedData
+
+        ap_sheet = output_file.sheets["AssetPrice"]
+        ae_sheet = output_file.sheets["Asset_and_Exchange"]
+        a_sheet = output_file.sheets["Asset"]
+
+        # First loop through, I want to know the total cost basis for all assets and exchanges after filtering out net cost under $0.01.
+        total_cost: RP2Decimal = RP2Decimal("0")
+        for asset, computed_data in asset_to_computed_data.items():
+            if not isinstance(asset, str):
+                raise RP2TypeError(f"Parameter 'asset' has non-string value {asset}")
+            ComputedData.type_check("computed_data", computed_data)
+
+            net_cost = RP2Decimal("0")
+            for current_transaction in computed_data.in_transaction_set:
+                in_xact = cast(InTransaction, current_transaction)
+                net_cost += in_xact.fiat_in_with_fee * (RP2Decimal("1") - computed_data.get_in_lot_sold_percentage(in_xact))
+
+            if net_cost < _FIAT_EMPTY_BAL_CUTOFF:
```

Comment:
The statement and others like it are now checking against ZERO rather than a cutoff value (exclude closed positions and empty wallets from the report). Originally I used cutoffs to get around #26 but I was able to munge my input data to prevent phantom crypto dust from adding an extraneous value to the output.